dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -199,4 +202,14 @@ static void throwIf(boolean condition, String message, int status) {
             throw new CustomWebApplicationException(message, status);
         }
     }
+
+    static boolean canDoAction(PermissionsInterface permissionsInterface, User user, Entry workflow, Role.Action action) {
+        try {
+            return workflow instanceof Workflow && permissionsInterface.canDoAction(user, (Workflow) workflow, action);
```

Comment:
It was like this, but you could do [this guard](https://github.com/dockstore/dockstore/blob/4aae4e244b8a156a747754d571e64f1178bb64ef/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L1000) and add a comment to the referenced ticket -- to avoid making unnecessary and potentially slow calls to SAM.